data/campaigns/campaignManager.ts

```diff
@@ -0,0 +1,111 @@
+// Campaign tracking
+
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import {CAMPAIGN_DATA_KEY} from 'data/asyncStorageKeys';
+import CAMPAIGNS, {CampaignId} from 'data/campaigns/campaigns';
+import {differenceInCalendarDays} from 'date-fns';
+import {logger} from 'logger';
+import * as Sentry from 'sentry-expo';
+import {z} from 'zod';
+
+const campaignViewSchema = z.object({
+  lastDisplayed: z.coerce.date().optional(),
+  timesDisplayed: z.number(),
+});
+
+type CampaignViewSchema = z.infer<typeof campaignViewSchema>;
+
+const campaignViewsSchema = z.record(campaignViewSchema);
+export type CampaignViewsSchema = z.infer<typeof campaignViewsSchema>;
+
+export interface ICampaignManager {
+  shouldShowCampaign(campaignId: CampaignId, atDate?: Date): boolean;
+  recordCampaignView(campaignId: CampaignId, atDate?: Date): Promise<void>;
+}
+
+class CampaignManager implements ICampaignManager {
+  private initialized = false;
+  private campaignViews: Record<CampaignId, CampaignViewSchema> = {} as Record<CampaignId, CampaignViewSchema>;
```

Comment:
@floatplane would it make sense to further index this by center ID? While I imagine we can deconflict names, more than one center having a campaign in the fourth quarter might like `campaign-q4-2023`

Reply:
overall I think it's best to keep it as a single string. they share a namespace with PostHog feature flags and Mixpanel tracking events (via `useCampaign`) so I would say we should deconflict names. now is probably a good time to do that, just to establish the pattern.

bigger problem is that we probably don't want `nwac-campaign-q4-2023` showing up when browsing Taos Avalanche Center. we could consider adding a list of allowed centers to the campaign definition, or I think you could do center filtering at the PostHog layer. having it explicit in code feels safer though.

Reply:
I guess different centers may want to see Mixpanel/Google Anayltics/whatever events flow into their _own_ dashboards? 🤔 there's going to be some thought required here.